gym_electric_motor/constraints.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+
+from .utils import set_state_array
+
+
+class Constraint:
+    """Base class for all constraints in the ConstraintMonitor."""
+
+    def __call__(self, state):
+        """Function that is called to check the constraint.
+
+        Args:
+            state(numpy.ndarray(float)): The current physical systems state.
+
+        Returns:
+              float in [0.0, 1.0]: Degree how much the constraint has been violated.
+                0.0: No violation
+                (0.0, 1.0): Undesired zone near to a full violation. No episode termination.
+                1.0: Full violation and episode termination.
+        """
+        raise NotImplementedError
+
+    def set_modules(self, ps):
+        """Called by the environment that the Constraint can read information from the PhysicalSystem.
+
+        Args:
+            ps(PhysicalSystem): PhysicalSystem of the environment.
+        """
+        pass
+
+
+class LimitConstraint(Constraint):
+    """Constraint to observe the limits on one or more system states.
+
+    This constraint observes if any of the systems state values exceeds the limit specified in the PhysicalSystem.
+
+    .. math::
+        1.0 <= s_i / s_{i,max}
+
+    For all :math:`i` in the set of PhysicalSystems states :math:`S`.
+
+    """
+
+    def __init__(self, observed_state_names='all'):
+        """
+        Args:
+            observed_state_names('all'/iterable(str)): The states to observe. \n
+                - 'all': Shortcut for observing all states.
+                - iterable(str): Pass an iterable containing all state names of the states to observe.
+        """
+        self._observed_state_names = observed_state_names
+        self._limits = None
+        self._observed_states = None
+
+    def __call__(self, state):
+        return float(any(state[self._observed_states] > 1.0))
+
+    def set_modules(self, ps):
+        self._limits = ps.limits
+        if self._observed_state_names == 'all':
+            self._observed_state_names = ps.state_names
+        if self._observed_state_names is None:
+            self._observed_state_names = []
+        self._observed_states = set_state_array(
+            dict.fromkeys(self._observed_state_names, 1), ps.state_names
+        ).astype(bool)
+
+
+class SquaredConstraint(Constraint):
```

Comment:
Shouldn't this inherit from LimitConstraint?

Reply:
There was no real reuse of code possible, imo. Maybe this becomes interesting if the number of available Contraints grows, but I don't expect that at the moment.